walletunlocker/service.go

```diff
@@ -119,6 +125,14 @@ func New(chainDir string, params *chaincfg.Params, noFreelistSync bool,
 	}
 }
 
+// Stop waits for all requests to be handled, then closes all channels in
+// preparation for the wrapping gRPC server to be shut down.
+func (u *UnlockerService) Stop() {
+	u.requestWg.Wait()
```

Comment:
Looking at this from a hight level it seems like it could have a possibility of deadlocking. Could we safely introduce a quit channel?